vumi_twilio_api/twilio_api.py

```diff
@@ -74,6 +193,32 @@ def __init__(self, message, format_='xml'):
         self.format_ = format_
 
 
+class Error(object):
+    """Error HTTP response object, returned for incorred API queries"""
+    def __init__(self, error_type, error_message):
+        self.error_type = error_type
+        self.error_message = error_message
+
+    @classmethod
+    def from_exception(cls, exception):
+        return cls(exception.__class__.__name__, exception.message)
+
+
+class Version(object):
+    """Version HTTP response object, returned for root resource"""
+    def __init__(self, name, uri, **kwargs):
+        self.Name = name
+        self.Uri = uri
+        self.SubresourceUris = kwargs
+
+
+class Call(object):
+    """Call HTTP response object, returned for the Calls resource"""
+    def __init__(self, **kwargs):
+        for key, value in kwargs.iteritems():
+            setattr(self, key, value)
```

Comment:
I'm really like the direction the new objects are headed in. I'm not that wild about using `__class__.__name__` for the name or `__dict__` for the values though. People don't expect the class name to matter much, and its easy for `__dict__` to end up with unexpected keys. Perhaps we could tweak things to structure things as:
```python
class Response(object):
    name = None
    def __init__(self, **kw):
        self._data = kw
    def format_xml(self):
        ...
    def format_json(self):
        ...

class Call(Response):
    name = 'Call'

class Version(Response):
    name = 'Version'
```
Thoughts? Like/not-like?